pkg/auth/userregistry/identitymapper/lookup.go

```diff
@@ -0,0 +1,34 @@
+package identitymapper
+
+import (
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	kuser "github.com/GoogleCloudPlatform/kubernetes/pkg/auth/user"
+
+	authapi "github.com/openshift/origin/pkg/auth/api"
+	identityregistry "github.com/openshift/origin/pkg/user/registry/identity"
+	"github.com/openshift/origin/pkg/user/registry/useridentitymapping"
+)
+
+type lookupIdentityMapper struct {
+	registry useridentitymapping.Registry
+}
+
+// NewLookupIdentityMapper returns a mapper that will look up existing mappings for identities
+func NewLookupIdentityMapper(registry useridentitymapping.Registry) authapi.UserIdentityMapper {
+	return &lookupIdentityMapper{registry}
+}
+
+// UserFor returns info about the user for whom identity info has been provided
+func (p *lookupIdentityMapper) UserFor(info authapi.UserIdentityInfo) (kuser.Info, error) {
+	name := identityregistry.IdentityName(info.GetProviderName(), info.GetProviderUserName())
+
+	mapping, err := p.registry.GetUserIdentityMapping(kapi.NewContext(), name)
```

Comment:
I think that's for API calls, or once you're within the registry... none of the places calling this have a context object either... would you want to create a new one in 8 places in order to call this method?